designs/vcenter-template-import.md

```diff
@@ -0,0 +1,91 @@
+# Import vcenter template before create
+
+## Introduction
+
+**Problem:** If a user doesn't have a template defined in the cluster spec, the create command 
+imports the latest available OVA for the specific version of the CLI. However, there is no way
+to import this OVA before running create without manually importing the OVA into a template
+in vcenter and specifying that template name in the cluster spec, even though we have that 
+automation logic in the CLI already. 
+
+### Goals and Objectives
+
+As an EKS Anywhere user:
+
+* I want to have the ability to pre-import the template before I run create
+* I want to specify the template I want to import in a simple manner
+* I want the create command to know to use the pre-imported template
+
+
+## Overview of Solution
+
+With this feature, a user can specify to import the latest OVA template before create by
+running a command. They would specify the cluster spec as input so that there is nothing 
+new that they need to know in terms of flags or different configuration. We would be introducing new
+subcommands as follows:
+
+```
+eksctl anywhere import vsphere template -f <cluster_spec.yaml>
+```
+
+
+### Solution Details
+
+With this feature, users will be able to import the latest OVA available based on the Kubernetes version
+and OS family defined in the cluster spec. We would introduce a new 
+`import` subcommand. Since this feature is specific to vsphere, we will introduce a `vsphere` 
+subcommand to give flexibility of the types of resources that we can import under 
+vsphere. This also gives us flexibility to import provider-agnostic resources under the `import`
+subcommand. Since this is importing the OVA into a vcenter template, we will be calling
+the final subcommand `template` based on the resource name. 
+
+Currently for the `create` subcommand, users can specify either a template if they already
+manually imported the OVA into a template in vcenter, or not specify the template field or leave the
+field empty if they want it imported via a generated name that is based on the Kubernetes
+version, OS, and hash of the build. 
+
+With this command, if a user has the template name defined, we will check to see if a template of that 
+name already exists and the appropriate tags are set. If the template does not exist, we will import the template under
+that name with the appropriate tags. If the template exists but the tags don't match the bundle 
```

Comment:
I think that is something that with the cluster spec we can loop through and document, so I wouldn't be concerned with the complexity. But I understand the lack of clarity and it would depend on who our end user is. I can add that section to this doc too, but I want to make sure we are in alignment with `import-images` or at least how they differ.

Reply:
Updated doc to include both types of users